domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -37,4 +40,25 @@ public async Task<ActionResult> Create([FromBody] CreateTransferAgreement reques
 
         return Created($"api/transfer-agreements/{result.Id}", result);
     }
+
+    [ProducesResponseType(typeof(TransferAgreementsResponse), 200)]
+    [ProducesResponseType(typeof(TransferAgreementsResponse), 404)]
+    [HttpGet("api/transfer-agreements")]
+    public async Task<ActionResult<TransferAgreementsResponse>> GetTransferAgreements()
+    {
+        var subject = User.FindSubjectClaim();
+        var userTin = User.FindTinClaim();
+
+        var transferAgreements = await transferAgreementRepository.GetTransferAgreementsList(Guid.Parse(subject), userTin);
+
+        var listResponse = transferAgreements.Select(ta => new TransferAgreementDto(
+            ta.Id,
+            ta.StartDate.ToUnixTimeSeconds(),
+            ta.EndDate.ToUnixTimeSeconds(),
+            ta.ReceiverTin))
+            .ToList();
+
+        return Ok(new TransferAgreementsResponse(listResponse));
```

Comment:
Use "204 No Content" if there is no agreements to return. Remember to add `[ProducesResponseType(204)]`

Reply:
added